### PR TITLE
test: convert the example in issue #300 into a test case

### PR DESCRIPTION
**Pull request by contestjia**
The new commit-id 16df11e can fix the issue

### CODE REVIEW

**Comment by contestjia** [comment]
@janisz 
 I add the example to test suite and make a new PR，Thanks

File: bigcache_test.go [hunk 1]
```diff
@@ -1132,3 +1132,29 @@ func TestCache_RepeatedSetWithBiggerEntry(t *testing.T) {
...
+// The new commit 16df11e change the encoding method,it can fix issue #300
+func TestBigCache_allocateAdditionalMemoryLeadPanic(t *testing.T){
+	t.Parallel()
+	cache, _ := NewBigCache(Config{
+		Shards:       1,
+		LifeWindow:   3 * time.Second,
+		MaxEntrySize: 52,
+	})
+	cache.Set("a",  blob(0xff, 235))
+	time.Sleep(2 * time.Second)
```
Comment (janisz) [inline bigcache_test.go:1146]: We should not sleep in tests. Please use `mockedClock`

Comment (contestjia) [reply bigcache_test.go]: I replaced time.Sleep with mockedClock.Please take a look

File: bigcache_test.go [hunk 2]
```diff
@@ -1132,3 +1132,37 @@ func TestCache_RepeatedSetWithBiggerEntry(t *testing.T) {
...
+	// reallocate memory
+	cache.Set("d",  blob(0xff, 1024))
+	ts += 4
+	clock.set(ts)
+	cache.Set("e",	blob(0xff, 3))
+	cache.Set("f", blob(0xff, 3))
+	cache.Set("g", blob(0xff, 3))
+	data,_ := cache.Get("g")
+	assertEqual(t, []byte{0xff,0xff,0xff}, data)
+}
```
Comment (janisz) [inline bigcache_test.go:1168]: Linter is failing due to `No newline at end of file`
```suggestion
}

```

**Comment by janisz** [comment]
@contestjia Coul you fix formatting to pass linter?

**Comment by contestjia** [comment]
@janisz yes, i fixed

File: bigcache_test.go [hunk 3]
```diff
@@ -1150,3 +1150,41 @@ func TestCache_RepeatedSetWithBiggerEntry(t *testing.T) {
...
+	clock := mockedClock{value: 0}
+	cache, _ := newBigCache(Config{
+		Shards:       1,
+		LifeWindow:   3 * time.Second,
+		MaxEntrySize: 52,
+	}, &clock)
+	ts := time.Now().Unix()
+	clock.set(ts)
+	cache.Set("a", blob(0xff, 235))
+	// mock time.Sleep(2 * time.Second)
```
[the remark this answers is not in the document]
Comment (janisz) [reply bigcache_test.go:1167]:
```suggestion

```

**Review by janisz — APPROVED** [review]

**Review by janisz — APPROVED** [review]

**Pull request merged**